File: backend/src/AccountBox.Data/Entities/Account.cs
using AccountBox.Core.Enums;
using System.ComponentModel.DataAnnotations;

namespace AccountBox.Data.Entities;

/// <summary>
/// Account 实体 - 表示网站上的一个账号
/// 密码以明文形式存储（适用于个人自托管场景）
/// </summary>
public class Account
{
    /// <summary>
    /// 主键
    /// </summary>
    public int Id { get; set; }

    /// <summary>
    /// 所属网站 ID（外键）
    /// </summary>
    public int WebsiteId { get; set; }

    /// <summary>
    /// 用户名
    /// </summary>
    public string Username { get; set; } = null!;

    /// <summary>
    /// 密码（明文存储）
    /// </summary>
    public string Password { get; set; } = null!;

    /// <summary>
    /// 备注（可为空）
    /// </summary>
    public string? Notes { get; set; }

    /// <summary>
    /// 标签（JSON 数组，用于分类和过滤）
    /// </summary>
    public string? Tags { get; set; }

    /// <summary>
    /// 账号状态（Active/Disabled）
    /// </summary>
    [Required]
    public AccountStatus Status { get; set; } = AccountStatus.Active;

    /// <summary>
    /// 扩展字段（JSON键值对，10KB限制）
    /// </summary>
    [Required]
    [MaxLength(10240, ErrorMessage = "扩展字段不能超过10KB")]
    public string ExtendedData { get; set; } = "{}";

    /// <summary>
    /// 软删除标志
    /// </summary>
    public bool IsDeleted { get; set; }

    /// <summary>
    /// 删除时间（软删除时记录）
    /// </summary>
    public DateTime? DeletedAt { get; set; }

    /// <summary>
    /// 创建时间
    /// </summary>
    public DateTime CreatedAt { get; set; }

    /// <summary>
    /// 最后更新时间
    /// </summary>
    public DateTime UpdatedAt { get; set; }

    /// <summary>
    /// 导航属性：所属网站
    /// </summary>
    public Website Website { get; set; } = null!;
}

File: frontend/src/types/account.ts
/**
 * 账号相关类型定义
 */

/**
 * 账号状态
 */
export type AccountStatus = 'Active' | 'Disabled'

/**
 * 账号实体（基础）
 */
export interface Account {
  id: number
  websiteId: number
  username: string
  password: string // 明文密码（2025-10-17架构变更：从加密存储切换为明文存储）
  notes?: string // 明文备注
  tags?: string
  status: AccountStatus // 账号状态
  extendedData?: Record<string, unknown> // 扩展字段（JSON 键值对）
  isDeleted: boolean
  deletedAt?: string
  createdAt: string
  updatedAt: string
}

/**
 * 账号响应（包含网站信息）
 */
export interface AccountResponse extends Account {
  websiteDomain: string
  websiteDisplayName?: string
}

/**
 * 创建账号请求
 */
export interface CreateAccountRequest {
  websiteId: number
  username: string
  password: string
  notes?: string
  tags?: string
  extendedData?: Record<string, unknown>
}

/**
 * 更新账号请求
 */
export interface UpdateAccountRequest {
  username: string
  password: string
  notes?: string
  tags?: string
  extendedData?: Record<string, unknown>
}

/**
 * 已删除账号响应（回收站）
 */
export interface DeletedAccountResponse extends AccountResponse {
  deletedAt: string
}
